packages/server/src/index.ts

```diff
@@ -9,28 +10,106 @@ import {
   takeSeat,
   leaveSeat,
 } from "./games";
+import {
+  createUserWithSessionId,
+  deleteUser,
+  getUser,
+  getUserBySessionId,
+} from "./users";
 import bodyParser from "body-parser";
 import cors from "cors";
 import path from "path";
 import { connectToDb } from "./db";
+import passport from "passport";
+import { Strategy as CustomStrategy } from "passport-custom";
 import {
   GameResponse,
   MovesType,
   DELETETHIS_getCurrentUser,
   User,
+  UserResponse,
 } from "@ogfcommunity/variants-shared";
 
 const LOCAL_ORIGIN = "http://localhost:3000";
 
+// passport.use(
+//   new LocalStrategy(async function (username, password, callback) {
+//     try {
+//       const user = await LocalUser.findOne({ name: username }).select(
+//         "+passwordHash"
+//       );
+//       // TODO: scrypt instead of bcrypt
+//       const passwordHash = user
+//         ? user.passwordHash
+//         : `$2b$${saltRounds}$TakeSomeTimeToCheckEvenIfUsernameIsWrong1234567890123`;
+
+//       if (!(await bcrypt.compare(password, passwordHash)) || !user) {
+//         return callback(null, false, {
+//           message: "Wrong username or password.",
+//         });
+//       }
+
+//       delete user._doc.passwordHash;
+//       return callback(null, user);
+//     } catch (err) {
+//       return callback(err);
+//     }
+//   })
+// );
+
+passport.use(
+  "guest",
+  new CustomStrategy(async function (req, callback) {
+    const token = req.session.id;
+    let user = await getUserBySessionId(token);
+    if (!user) {
+      user = await createUserWithSessionId(token);
+    }
+    return callback(null, user);
+  })
+);
+
+passport.serializeUser<string>(function (user: UserResponse, callback) {
+  callback(null, user.id);
+});
+
+passport.deserializeUser<string>(function (id, callback) {
+  getUser(id)
+    .then((user) => {
+      if (user) {
+        callback(null, user);
+      } else {
+        callback(new Error(`No user with ID: ${id}`));
+      }
+    })
+    .catch((err) => {
+      callback(err);
+    });
+});
+
 // initialize Express
 const app = express();
 app.use(
   bodyParser.urlencoded({
     extended: true,
   })
 );
-app.use(bodyParser.json());
-app.use(cors({ origin: LOCAL_ORIGIN, credentials: true }));
+app.use(bodyParser.json()); // TODO: app.use(express.json()) instead? Difference?
+app.use(cors({ origin: LOCAL_ORIGIN, credentials: true })); // TODO: Is this still necessary with dev proxy?
+app.use(
+  session({
+    // TODO: Cookie banner or permission necessary?
+    secret: process.env.SESSION_SECRET || "Corybas aconitiflorus",
```

Comment:
> **secret**
>
> **Required option**
>
>This is the secret used to sign the session ID cookie. This can be either a string for a single secret, or an array of multiple secrets. If an array of secrets is provided, only the first element will be used to sign the session ID cookie, while all the elements will be considered when verifying the signature in requests. The secret itself should be not easily parsed by a human and would best be a random set of characters. A best practice may include:
>
>* The use of environment variables to store the secret, ensuring the secret itself does not exist in your repository. Periodic updates of the secret, while ensuring the previous secret is in the array.
>* Using a secret that cannot be guessed will reduce the ability to hijack a session to only guessing the session ID (as determined by the genid option).
>
>Changing the secret value will invalidate all existing sessions. In order to rotate the secret without invalidating sessions, provide an array of secrets, with the new secret as first element of the array, and including previous secrets as the later elements.
https://www.npmjs.com/package/express-session